src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Unix.cs

```diff
@@ -267,6 +267,20 @@ protected override void Dispose(bool disposing)
                         // name will be removed immediately.
                         Interop.Sys.Unlink(_path); // ignore errors; it's valid that the path may no longer exist
                     }
+
+                    // Closing the file handle can fail, e.g. due to out of disk space
+                    // Throw these errors as exceptions when disposing
+                    if (_fileHandle != null && !_fileHandle.IsClosed && disposing)
+                    {
+                        _fileHandle.t_lastCloseErrorInfo = null;
+
+                        _fileHandle.Dispose();
+
+                        if (_fileHandle.t_lastCloseErrorInfo != null)
+                        {
+                            throw Interop.GetExceptionForIoErrno(_fileHandle.t_lastCloseErrorInfo.GetValueOrDefault(), _path, isDirectory: false);
+                        }
+                    }
```

Comment:
I'm having some trouble getting this to run in WSL so I submitted this change to check the result in the CI.

Reply:
OK my WSL is working now, and I was able to write a small test to verify that this code is being reached successfully when expected.